### DISCUSSION TITLE
How to read attribute values in a node's dependant branch and not only in their children?

### DISCUSSION THREAD

**Discussion by Illegit** [freeplane/freeplane#1765] [Help] [answered]
Hi, I need help to correct the following formula: =format(node.children.collect{it['Date'].to.date}.min(),'yyyy-MM-dd')
It shows the min date in the attribute called "Date" in  some of the children of a node. Which one is the kind of formula or solution that I have to adopt when I need to read the attribute in the children nodes but even in their respective children and so on?
In this example you see that I'm reading only the children but not the other nodes and the result must be: 2024-02-02, and not 2024-03-15.
![image](https://github.com/freeplane/freeplane/assets/62674157/9b0c6f9e-70a9-41d1-a81f-13e1817237e2)

Thanks

**Comment by macmarrum** [comment, marked as answer]
You can swap `children` for `findAll().drop(1)`, i.e.

```groovy
=format(node.findAll().drop(1).collect { it['Date'].date }.min(), 'yyyy-MM-dd')
```

- `findAll()` returns all descendants, including the starting node
- `drop(1)` removes the starting node from the collection

**Comment by Illegit** [comment]
Thanks macmarrum. That meets my need.
